src/Pos/Util/LogSafe.hs

```diff
@@ -0,0 +1,66 @@
+-- | Safe/secure logging
+
+module Pos.Util.LogSafe
+       ( SecureLogWrapped(..)
+       , logDebugS
+       , logInfoS
+       , logNoticeS
+       , logWarningS
+       , logErrorS
+       ) where
+
+import           Control.Monad.Trans (MonadTrans)
+import           Data.List           (isSuffixOf)
+import           System.Wlog         (CanLog (..), HasLoggerName (..), Severity (..),
+                                      loggerName)
+import           System.Wlog.Logger  (logMCond)
+import           Universum
+
+
+newtype SecureLogWrapped m a = SecureLogWrapped
+    { getSecureLogWrapped :: m a
+    } deriving (Functor, Applicative, Monad, MonadIO)
+
+instance MonadTrans SecureLogWrapped where
+    lift = SecureLogWrapped
+
+instance (MonadIO m) => CanLog (SecureLogWrapped m) where
+    dispatchMessage
+        (loggerName      -> name)
+        severity
+        msg =
+      let acceptable p
+              | "fileHandler" `isPrefixOf` p ||
```

Comment:
Strings is not the way to determine whether something is public or not. Should be a sum type.